### DISCUSSION TITLE
System.AccessViolationException Direct2D1

### DISCUSSION THREAD

**Discussion by Felicitas12** [amerkoleci/Vortice.Windows#403] [Q&A] [answered]
An unhandled exception of type 'System.AccessViolationException' occurred in Vortice.Direct2D1.dll
Attempted to read or write protected memory. This is often an indication that other memory is corrupt.

System.AccessViolationException: 'Attempted to read or write protected memory. This is often an indication that other memory is corrupt.'

I am trying to create _renderTarget for Direct2D to integarte it in Directx12 project.
`_renderTarget = _d2dFactory.CreateDxgiSurfaceRenderTarget(new IDXGISurface(_backbufferIndex), new RenderTargetProperties());` This line causes an exception.
I'm very new to this, so I'd appreciate it if someone could help me with this.

**Comment by amerkoleci** [comment]
Give some code so I can help, this way is useless.

**Reply by Felicitas12** [reply]
https://pastebin.com/mb7fJTfZ

**Reply by Felicitas12** [reply]
DrawFrame, 685 line

**Reply by amerkoleci** [reply, marked as answer]
You should use D3D12On11, follow this sample https://github.com/microsoft/DirectX-Graphics-Samples/blob/master/Samples/Desktop/D3D1211On12/src/D3D1211On12.cpp

**Reply by amerkoleci** [reply]
I'll add an example in the Vortice.Windows.Samples repo once I have some time!

**Reply by Felicitas12** [reply]
D2d does not working without D3D12On11?

**Reply by Felicitas12** [reply]
Would this example work in my case? 
https://learn.microsoft.com/en-us/windows/win32/direct2d/direct2d-and-direct3d-interoperation-overview 
I originally tried to do by this example, but ran into a problem, which is actually what I opened the discussion about

**Reply by Felicitas12** [reply]
Or this https://github.com/Microsoft/DirectX-Graphics-Samples/blob/master/MiniEngine/Core/TextRenderer.cpp
